gocd/pipelines/relay.yaml

```diff
@@ -0,0 +1,118 @@
+# More information on gocd-flavor YAML can be found here:
+# - https://github.com/tomzo/gocd-yaml-config-plugin#pipeline
+# - https://www.notion.so/sentry/GoCD-New-Service-Quickstart-6d8db7a6964049b3b0e78b8a4b52e25d
+format_version: 10
+pipelines:
+    deploy-relay:
+        environment_variables:
+            GCP_PROJECT: internal-sentry
+            GKE_CLUSTER: zdpwkxst
+            GKE_REGION: us-central1
+            GKE_CLUSTER_ZONE: b
+            GKE_BASTION_ZONE: b
+        group: relay
+        lock_behavior: unlockWhenFinished
+        materials:
+            relay_repo:
+                git: git@github.com:getsentry/relay.git
+                shallow_clone: true
+                branch: master
+                destination: relay
+        stages:
+            - checks:
+                  approval:
+                      type: manual
+                  fetch_materials: true
+                  jobs:
+                      checks:
+                          environment_variables:
+                              # Required for checkruns.
+                              GITHUB_TOKEN: "{{SECRET:[devinfra-github][token]}}"
+                          timeout: 1800
+                          elastic_profile_id: relay
+                          tasks:
+                              - script: |
+                                    /devinfra/scripts/checks/githubactions/checkruns.py \
+                                    getsentry/relay \
+                                    ${GO_REVISION_RELAY_REPO} \
+                                    "Integration Tests" \
+                                    "Test (macos-latest)" \
+                                    "Test (windows-latest)" \
+                                    "Test All Features (ubuntu-latest)" \
+                                    "self-hosted-end-to-end" \
+                                    "Sentry-Relay Integration Tests" \
+                                    "Test Python 3.8"
```

Comment:
Did we have these listed in freight before? We can remove "Test Python 3.8" from this list, and should add "Push GCR Docker Image"

Reply:
It's currently this in Freight, I'll update it.

```
          "Integration Tests",
          "Test (macos-latest)",
          "Test (windows-latest)",
          "Test All Features (ubuntu-latest)",
          "Push GCR Docker Image"
```

Reply:
Any reason not to have `"Sentry-Relay Integration Tests"` and `"self-hosted-end-to-end"`?

Reply:
I think we skipped those because they were often slow and unstable. The situation has improved since, so we can consider to add them. Feel free to defer this to a later time.

Reply:
Ok - will leave that up to the ingest team.